src/ui/hooks/useChaptersTrack.ts

```diff
@@ -0,0 +1,39 @@
+import { useCallback, useContext, useSyncExternalStore } from 'react';
+import { PlayerContext } from '@theoplayer/react-native-ui';
+import { PlayerEventType, type PlayerEventMap, TextTrack } from 'react-native-theoplayer';
+
+const TEXT_TRACK_CHANGE_EVENTS = [PlayerEventType.LOADED_DATA, PlayerEventType.TEXT_TRACK_LIST] satisfies ReadonlyArray<keyof PlayerEventMap>;
+
+/**
+ * Retain first chapters track encountered in the textTracks list.
+ */
+export function filterChaptersTrack(textTracks: TextTrack[] | undefined): TextTrack | undefined {
+    return textTracks && textTracks.find(isChaptersTrack);
+  }
+  
+  /**
+   * Query whether a track is a valid chapters track.
+   */
+  export function isChaptersTrack(textTrack: TextTrack | undefined): boolean {
+    return !!textTrack && (textTrack.kind === 'chapters');
+  }
```

Comment:
We shouldn't have to deal with an `undefined` input here.
```suggestion
  export function isChaptersTrack(textTrack: TextTrack): boolean {
    return textTrack.kind === 'chapters';
  }
```